simple_rpc/simple_rpc.py

```diff
@@ -184,3 +142,78 @@ def call_method(self, name, *args):
         if method['return']['fmt']:
             return self._read(method['return']['fmt'])
         return None
+
+
+class SerialInterface(Interface):
+    def __init__(self, device, baudrate=9600, wait=2, autoconnect=True):
+        """
+        :arg str device: Device name.
+        :arg int baudrate: Baud rate.
+        :arg int wait: Time in seconds before communication starts.
+        :arg bool autoconnect: Automatically connect.
+        """
+        super().__init__(device, baudrate)
+        self._wait = wait
+
+        if autoconnect:
+            self.open()
+
+    def is_open(self):
+        """Query interface state."""
+        return self._connection.isOpen()
+
+    def open(self):
+        """Connect to device."""
+        self._open()
+        sleep(self._wait)
+        super().open()
+
+    def close(self):
+        """Disconnect from device."""
+        super().close()
+        self._close()
+
+
+class SocketInterface(Interface):
+    def __init__(self, device, baudrate=9600):
+        """
+        :arg str device: Device name.
+        :arg int baudrate: Baud rate.
+        """
+        super().__init__(device, baudrate)
+
+        self._get_methods()
```

Comment:
I called `_get_methods` directly here. At first I wrote this using autoconnect, open, and close but decided to do it this way since you'd have to use private method calls to do it correctly.